protocols/kad/src/query/peers/closest/disjoint.rs

```diff
@@ -137,14 +137,12 @@ impl ClosestDisjointPeersIter {
     pub fn next(&mut self, now: Instant) -> PeersIterState {
         let mut state = None;
 
-        // Order in which to query the iterators to ensure fairness. Make sure
-        // to query the previously queried iterator last.
-        let iter_order = {
-            let mut all = (0..self.iters.len()).map(Into::into).collect::<Vec<_>>();
-            let mut next_up = all.split_off((self.last_queried + 1).into());
-            next_up.append(&mut all);
-            next_up
-        };
+        // Order in which to query the iterators to ensure fairness.
+        let iter_order = (0..self.iters.len()).map(Into::into).cycle()
+            // Make sure to query the previously queried iterator last.
+            .skip(Into::<usize>::into(self.last_queried) + 1)
+            // Query each iterator at most once.
+            .take(self.iters.len());
```

Comment:
This is not quite what I had in mind, I was thinking about getting rid of `last_queried` and instead keep the `Cycle` iterator for the indices as a local property.

Reply:
Sorry for being a bit slow to comprehend here. 

> Do you see a problem with that?

Only thing that comes to my mind is that one has to make sure to query each iterator at most once.

What do you think of https://github.com/libp2p/rust-libp2p/pull/1473/commits/4c8fe6e6adf933200ade85b878dcde9b1c5c5df1 ?

Reply:
Looks good, I only left small suggestions for simplification.